### PR TITLE
Refactor-api_error

### PR DESCRIPTION
**Pull request by Jeonghyun-Kim**
- add package http-status-codes
- strictly typed ERRORs

### CODE REVIEW

File: src/defines/errors.ts [hunk 1]
```diff
@@ -1,40 +1,74 @@
...
-    new CustomError('CE004', 401, 'Token expired', message ?? 'Token expired'),
-} as const;
+  static isApiError(x: any): x is ApiError {
+    return (
+      x.code !== undefined &&
+      x.statusCode !== undefined &&
+      x.name !== undefined &&
+      x.message !== undefined
+    );
+  }
```
Comment (kjsik11) [inline src/defines/errors.ts:64]:
```suggestion
  static isApiError(x: any): x is ApiError {
    return (
      x.code !== undefined &&
      x.statusCode !== undefined &&
      x.name !== undefined &&
      x.message !== undefined
    );
  }
```

How about use "err" or "error" instead "x"

**Review by kjsik11 — COMMENTED** [review]
gooood😄

File: src/defines/errors.ts (hunk 1 above)
Comment (Jeonghyun-Kim) [reply src/defines/errors.ts]: resolved by https://github.com/coxwave/nextjs-template/commit/aa44019d9ea762d4c96449d38291905dbf7c63ea

**Review by kjsik11 — APPROVED** [review]
Goood

**Pull request merged**